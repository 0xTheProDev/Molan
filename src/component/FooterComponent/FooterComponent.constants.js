// @flow
import { SERVICE_STATUS } from 'constants/services';
import { CopyRightHolderItem, ServiceStatus } from './FooterComponent.flow';

export const COPYRIGHT_HOLDERS: Array<CopyRightHolderItem> = Object.freeze([
  {
    title: 'Bytes Club',
    url: 'https://github.com/BytesClub',
  },
  {
    title: 'Progyan Bhattacharya',
    url: 'https://progyan1997.github.io',
  }
]);


const SERVICE_STATUS_LOADING: ServiceStatus = Object.freeze({
  icon: 'question',
  color: 'yellow',
  label: 'Loading',
});

const SERVICE_STATUS_READY: ServiceStatus = Object.freeze({
  icon: 'check',
  color: 'green',
  label: 'Ok',
});

const SERVICE_STATUS_ERROR: ServiceStatus = Object.freeze({
  icon: 'close',
  color: 'red',
  label: 'Failed',
});

export const SERVICE_STATUS_OPTIONS = Object.freeze({
  LOADING: SERVICE_STATUS_LOADING,
  READY: SERVICE_STATUS_READY,
  ERROR: SERVICE_STATUS_ERROR,
});

export const SERVICE_STATUS_MAPPING = Object.freeze({
  [SERVICE_STATUS.LOADING]: SERVICE_STATUS_LOADING,
  [SERVICE_STATUS.READY]: SERVICE_STATUS_READY,
  [SERVICE_STATUS.ERROR]: SERVICE_STATUS_ERROR,
});
