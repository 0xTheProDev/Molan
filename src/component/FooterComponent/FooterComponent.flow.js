// @flow
export type FooterComponentProps = {
  ServiceStatus: number,
  reloadServiceStatus: () => void,
};

export type CopyRightHolderItem = {
  title: string,
  url: string,
};

export type ServiceStatus = {
  icon: string,
  color: string,
  label: string,
};

export type FooterComponentInternalProps = {
  CopyRightTitleText: string,
  CopyRightHolders: Array<CopyRightHolderItem>,
  ServiceStatusTitleText: string,
  ServiceStatus: ServiceStatus,
  reloadServiceStatus: () => void,
};
