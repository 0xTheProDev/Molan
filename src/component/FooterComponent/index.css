.footer-content {
    height: 60px !important;
    background-color: #FFF8DC !important;
}

.service-status {
    border: 1px solid rgba(34,36,38,.15) !important;
    border-radius: 10px !important;
    cursor: pointer !important;
}

.service-status:hover {
    border-color: #000 !important;
}

.service-status:active, .service-status:focus {
    border-color: #85b7d9 !important;
}

.copyright {
    float: left;
}

.footer-cols {
    float: right;
}

@media (min-width: 600px) {
    .service-status {
        padding: 5px 0 !important;
        margin: 0 180px !important;
    }

    .footer-cols {
        width: 40%;
    }
}

@media (max-width: 599px) {
    .service-status {
        padding: 5px !important;
    }
}