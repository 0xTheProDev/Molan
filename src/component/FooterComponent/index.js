import React, { Component } from "react";
import { Grid, Segment, Divider } from "semantic-ui-react";
import "./index.css";

export default class FooterComponent extends Component {
    render() {
        return (
            <div className="footer-content">
                <div className="copyright">
                    <Segment basic>
                        &copy; All rights reserved by <a href="https://github.com/Tech-Mantra" target="_blank" rel="noopener noreferrer">Tech-Mantra</a>
                    </Segment>
                </div>
                <div className="footer-cols">
                    <Grid columns={3} relaxed>
                        <Grid.Column>
                          <Segment basic>About</Segment>
                        </Grid.Column>
                        <Divider vertical hidden/>
                        <Grid.Column>
                          <Segment basic>Privacy Policy</Segment>
                        </Grid.Column>
                        <Divider vertical hidden/>
                        <Grid.Column>
                          <Segment basic>Terms of Usage</Segment>
                        </Grid.Column>
                    </Grid>
                </div>
            </div>
        );
    }
}
