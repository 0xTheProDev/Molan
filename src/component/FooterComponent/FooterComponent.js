// @flow
import React from 'react';
import { Button, Grid, Icon, Segment } from 'semantic-ui-react';
import { noop } from 'util/common';

import { FooterComponentInternalProps } from './FooterComponent.flow';
import { COPYRIGHT_HOLDERS, SERVICE_STATUS_OPTIONS } from './FooterComponent.constants';
import Styles from './FooterComponent.module.css';

export function CopyRightSection({ CopyRightTitleText, CopyRightHolders }: FooterComponentInternalProps) {
  return (
    <div className={Styles['copyright']} role="contentinfo">
      <Segment basic>
        <i>&copy;</i>
        <span className="lg-only">{CopyRightTitleText}</span>
        {
          CopyRightHolders.map((copyRightHolder) =>
            <a key={copyRightHolder.url} href={copyRightHolder.url} target="_blank" rel="noopener noreferrer">
              {copyRightHolder.title}
            </a>
          )
        }
      </Segment>
    </div>
  );
}

export function ServiceStatusSection({ ServiceStatus, ServiceStatusTitleText, reloadServiceStatus }: FooterComponentInternalProps) {
  return (
    <Grid columns={1} centered relaxed>
      <Grid.Column>
        <Segment basic textAlign="center">
          <Button className={Styles['service-status']} onClick={reloadServiceStatus} role="status">
            <Icon
              name={ServiceStatus.icon}
              color={ServiceStatus.color}
              aria-label={ServiceStatus.label}
              aria-hidden={ServiceStatus.label}
            />
            {ServiceStatusTitleText}
          </Button>
        </Segment>
      </Grid.Column>
    </Grid>
  );
}

export function FooterComponent(props: FooterComponentInternalProps) {
  return (
    <div className={Styles['footer-content']}>
      <CopyRightSection {...props} />
      <div className={Styles['footer-cols']}>
        <ServiceStatusSection {...props} />
      </div>
    </div>
  );
}

FooterComponent.defaultProps = {
  CopyRightTitleText: 'All rights reserved by',
  CopyRightHolders: COPYRIGHT_HOLDERS,
  ServiceStatusTitleText: 'Service Status',
  ServiceStatus: SERVICE_STATUS_OPTIONS.LOADING,
  reloadServiceStatus: noop,
};
