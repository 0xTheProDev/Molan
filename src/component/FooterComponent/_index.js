// @flow
import React from 'react';

import { FooterComponentProps } from './FooterComponent.flow';
import { FooterComponent } from './FooterComponent';
import { SERVICE_STATUS_MAPPING } from './FooterComponent.constants';

export default function FooterComponentFactory({ ServiceStatus, reloadServiceStatus }: FooterComponentProps) {
  const ServiceStatusOption = SERVICE_STATUS_MAPPING[ServiceStatus];

  return (
    <FooterComponent
      ServiceStatus={ServiceStatusOption}
      reloadServiceStatus={reloadServiceStatus}
    />
  );
}

export { FooterComponentProps, FooterComponentFactory as FooterComponent };
