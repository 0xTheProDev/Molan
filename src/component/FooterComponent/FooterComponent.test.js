import React from 'react';

import { FooterComponent } from './FooterComponent';

describe('Footer Component', () => {
  let
    originalAddEventListener,
    addEventListenerStub,
    getStatusStub,
    minProps;

  beforeAll(() => {
    originalAddEventListener = global.addEventListener
    addEventListenerStub = jest.fn();
    global.addEventListener = addEventListenerStub;

    getStatusStub = jest.fn();
    minProps = {
      service: {},
      getStatus: getStatusStub,
    };
  });

  afterAll(() => {
    global.addEventListener = originalAddEventListener;
  });

  it('should render', () => {
    const wrapper = shallow(<FooterComponent {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show service status when service is up', () => {
    const diffProps = {
      service: {
        status: true,
      },
    };
    const wrapper = shallow(<FooterComponent {...minProps} {...diffProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show service status when service is up', () => {
    const diffProps = {
      service: {
        status: false,
      },
    };
    const wrapper = shallow(<FooterComponent {...minProps} {...diffProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should listen to network status', () => {
    const _wrapper = shallow(<FooterComponent {...minProps} />);
    expect(addEventListenerStub).toBeCalledWith('online', expect.any(Function));
    expect(addEventListenerStub).toBeCalledWith('offline', expect.any(Function));
  });

  it('should refetch service status when clicked on icon', () => {
    const wrapper = shallow(<FooterComponent {...minProps} />);
    wrapper.find('.service-status').simulate('click');
    expect(getStatusStub).toHaveBeenCalled();
  });
});
