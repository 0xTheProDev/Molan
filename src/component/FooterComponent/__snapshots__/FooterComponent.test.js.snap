// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render 1`] = `
<div
  className="footer-content"
>
  <div
    className="copyright"
  >
    <Segment
      basic={true}
    >
      © 
      <span
        className="lg-only"
      >
        All rights reserved by
      </span>
        
      <a
        href="https://github.com/Tech-Mantra"
        rel="noopener noreferrer"
        target="_blank"
      >
        Tech-Mantra
      </a>
      ,   
      <a
        href="https://github.com/BytesClub"
        rel="noopener noreferrer"
        target="_blank"
      >
        Bytes Club
      </a>
    </Segment>
  </div>
  <div
    className="footer-cols"
  >
    <Grid
      centered={true}
      columns={1}
      relaxed={true}
    >
      <GridColumn>
        <Segment
          basic={true}
          textAlign="center"
        >
          <div
            className="service-status"
            onClick={[Function]}
          >
            <Icon
              aria-hidden="Loading"
              aria-label="Loading"
              as="i"
              color="yellow"
              name="question"
            />
            Service Status
          </div>
        </Segment>
      </GridColumn>
    </Grid>
  </div>
</div>
`;

exports[`Footer Component should show service status when service is up 1`] = `
<div
  className="footer-content"
>
  <div
    className="copyright"
  >
    <Segment
      basic={true}
    >
      © 
      <span
        className="lg-only"
      >
        All rights reserved by
      </span>
        
      <a
        href="https://github.com/Tech-Mantra"
        rel="noopener noreferrer"
        target="_blank"
      >
        Tech-Mantra
      </a>
      ,   
      <a
        href="https://github.com/BytesClub"
        rel="noopener noreferrer"
        target="_blank"
      >
        Bytes Club
      </a>
    </Segment>
  </div>
  <div
    className="footer-cols"
  >
    <Grid
      centered={true}
      columns={1}
      relaxed={true}
    >
      <GridColumn>
        <Segment
          basic={true}
          textAlign="center"
        >
          <div
            className="service-status"
            onClick={[Function]}
          >
            <Icon
              aria-hidden="Loading"
              aria-label="Loading"
              as="i"
              color="yellow"
              name="question"
            />
            Service Status
          </div>
        </Segment>
      </GridColumn>
    </Grid>
  </div>
</div>
`;

exports[`Footer Component should show service status when service is up 2`] = `
<div
  className="footer-content"
>
  <div
    className="copyright"
  >
    <Segment
      basic={true}
    >
      © 
      <span
        className="lg-only"
      >
        All rights reserved by
      </span>
        
      <a
        href="https://github.com/Tech-Mantra"
        rel="noopener noreferrer"
        target="_blank"
      >
        Tech-Mantra
      </a>
      ,   
      <a
        href="https://github.com/BytesClub"
        rel="noopener noreferrer"
        target="_blank"
      >
        Bytes Club
      </a>
    </Segment>
  </div>
  <div
    className="footer-cols"
  >
    <Grid
      centered={true}
      columns={1}
      relaxed={true}
    >
      <GridColumn>
        <Segment
          basic={true}
          textAlign="center"
        >
          <div
            className="service-status"
            onClick={[Function]}
          >
            <Icon
              aria-hidden="Loading"
              aria-label="Loading"
              as="i"
              color="yellow"
              name="question"
            />
            Service Status
          </div>
        </Segment>
      </GridColumn>
    </Grid>
  </div>
</div>
`;
